osgbuild/importer.py

```diff
@@ -152,9 +153,9 @@ def make_source_file(url, cached_filename, upstream_dir, provider=None):
     source_filename = os.path.join(upstream_dir, provider+".srpm.source")
 
     source_contents = """\
-%(cached_filename)s
-# Downloaded from '%(url)s'
-""" % locals()
+{0}{1}
+# Downloaded from {2}
+""".format(cached_filename, " sha1sum="+sha1sum if sha1sum else "", url)
```

Comment:
Also, this inline-if-else in the middle of a list of args is a bit awkward for me to read.

If you like it as you have it, fine, but you could also compute it on a separate line.  Eg,

```
if want_sha1sum:
    srcspec = "{cached_filename} sha1sum={sha1sum}".format(**locals())
else:
    srcspec = cached_filename
```

Reply:
lol I did the split version first but I thought you'd like the inline version better

Reply:
Garsh, well thanks then! ;)

I like inline if-else things if its the sole argument, eg `x = this if whatever else that` or `return thing if good else None`, but
1. in a list it's a bit harder to read, and
2. even alone, `" sha1sum="+sha1sum if sha1sum else ""` is somewhat hard for me visually to parse, because i have to think harder about whether it means `(" sha1sum="+sha1sum) if sha1sum else ""` or `" sha1sum="+(sha1sum if sha1sum else "")`